localstack/http/proxy.py

```diff
@@ -113,6 +113,8 @@ def _copy_request(
     :param headers: optional headers to overwrite
     :return: a new request with slightly modified underlying environment but the same data stream
     """
+    # ensure that the headers in the env are set on the environment
+    set_environment_headers(request.environ, request.headers)
```

Comment:
my only concern here would be that the environment will still contain headers that were potentially removed in `request.headers` earlier. but not sure this is really a problem.